From 57bbffda1293dd045b4c2ff034e9a5bd826ba3e9 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 28 Jan 2021 13:35:03 +0100
Subject: [PATCH] Fix atomic modesetting with musl

atomic modesetting seems to fail with drm weston backend and this patch fixes
it, below errors are seen before weston exits

atomic: couldn't commit new state: Invalid argument

Upstream-Status: Submitted [https://gitlab.freedesktop.org/wayland/weston/-/issues/158]
Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 libweston/backend-drm/kms.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libweston/backend-drm/kms.c b/libweston/backend-drm/kms.c
index ee1c7fa5..8aae2617 100644
--- a/libweston/backend-drm/kms.c
+++ b/libweston/backend-drm/kms.c
@@ -1204,8 +1204,8 @@ drm_pending_state_apply_atomic(struct drm_pending_state *pending_state,
 		wl_list_for_each(plane, &b->plane_list, link) {
 			drm_debug(b, "\t\t[atomic] starting with plane %lu disabled\n",
 				  (unsigned long) plane->plane_id);
-			plane_add_prop(req, plane, WDRM_PLANE_CRTC_ID, 0);
-			plane_add_prop(req, plane, WDRM_PLANE_FB_ID, 0);
+			//plane_add_prop(req, plane, WDRM_PLANE_CRTC_ID, 0);
+			//plane_add_prop(req, plane, WDRM_PLANE_FB_ID, 0);
 		}
 
 		flags |= DRM_MODE_ATOMIC_ALLOW_MODESET;
